domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt

```diff
@@ -722,7 +722,7 @@ class ProfileManagementController @Inject constructor(
   }
 
   private fun onlyLetters(name: String): Boolean {
-    return name.matches(Regex("^[ A-Za-z]+\$"))
+    return name.matches(Regex("^(?!.*^[-.'])^(?!.*[.'-]\$).*^(?!.*[0-9!@#\$%^&*()_+={}\\[\\]|\\\\;\",?/<>`~]).*[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'-]"))
   }
```

Comment:
Please add tests for various names that should be prohibited/allowed to ensure the pattern is working as expected.

Reply:
Hello, @srushtirk , could you give some test cases/ examples of what names are acceptable and which ones are not acceptable?
What I want to make sure is that:
Although you have allowed `.`, `-` and `'` in the name, can the name start/end with one of these symbols?(For ex, is `-jishnu..` allowed?)

This is one of the constraints I found, but I want to know if there are any more of such constraints that are needed to be taken care of. 
PTAL @srushtirk Thanks!